Extra dependency `asyncio` included

[`asyncio`](https://pypi.org/project/asyncio/) is a dependency only for upto Python 3.4. Since the versions of Python supported are 3.7+, it doesn't seem to be required, but [is included](https://github.com/twilio/twilio-python/blob/9fd5f3f266c2e4172ed951a52640e285065a0501/setup.py#L27).

This is causing issues with running `pip check` in https://github.com/conda-forge/twilio-feedstock/pull/94, which I help maintain.

### Technical details:
* twilio-python version: 8.0.0
* python version: 3.7+

`asyncio` has broken our Serverless Lambda function(Python v3.8):
> [ERROR] SyntaxError: invalid syntax (base_events.py, line 296)
...
  File "/var/task/django/utils/deprecation.py", line 1, in <module>
    import asyncio
  File "/var/task/asyncio/__init__.py", line 21, in <module>
    from .base_events import *
    
As a workaround fixed using version lock with previous version  of Twilio

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,6 +2,5 @@ pygments>=2.7.4 # not directly required, pinned by Snyk to avoid a vulnerability
 pytz
 requests>=2.0.0
 PyJWT>=2.0.0, <3.0.0
-asyncio>=3.4.3
 aiohttp>=3.8.4
 aiohttp-retry>=2.8.3

--- a/setup.py
+++ b/setup.py
@@ -24,7 +24,6 @@ setup(
         "pytz",
         "requests >= 2.0.0",
         "PyJWT >= 2.0.0, < 3.0.0",
-        "asyncio>=3.4.3",
         "aiohttp>=3.8.4",
         "aiohttp-retry>=2.8.3",
     ],

